src/db/sysdb_ops.c

```diff
@@ -4909,7 +4909,9 @@ errno_t sysdb_search_object_by_id(TALLOC_CTX *mem_ctx,
     if (filter == NULL) {
         return ENOMEM;
     }
-
+    if (id > 20000 || id < 0) {
```

Comment:
Why 20000 was chosen? Please, add a new #define with a meaningful name (or check if we already have one and use it) in order to have things more clear.

Reply:
will do

Reply:
But why 20000 was chosen? I don't understand where the 20000 comes from.

Reply:
I've checked the code and seems that the check you'd like to do is based on domain.
So, you'd have to do something like (not tested):
```
if ((domain->max_id != 0 && id > domain->max_id) ||  id < domain->min_id ) {
...
}
```

The reason for this check is because:
- min_id and max_id may be defined by domain through sssd.conf
- defining max_id = 0 means there's no limit for the max_id

Reply:
Last but not least, I'd expect at least some debug message added warning that min_id/max_id conditions are not match.

Reply:
Also, I'd like to hear from @pbrezina whether he thinks would be better to do this check at the sysdb level or at the cache_req level.